### DISCUSSION TITLE
Difference in API output

### DISCUSSION THREAD

**Discussion by meb427** [open-meteo/open-meteo#455] [Show and tell]
Hi,

I have setup 2 instances that pull all the forecasts using the advised cron. I created a small script to compare the data VS the openmeteo own API. I was unable to match the result for temperature_2m on 3 API's forecast/dwd-icon/ecmwf.

What I noticed is strange the openmeto api is some time +0.1c on full dataset or -0.1c

Example

Data for endpoint: v1/forecast

     https://api.open-meteo.com_v1/forecast  http://192.168.1.114:8080_v1/forecast  http://192.168.1.110:8080_v1/forecast     Match
0                                      20.9                                   21.0                                   21.0  No Match
1                                      20.8                                   20.9                                   20.9  No Match
2                                      20.9                                   20.9                                   20.9     Match
3                                      20.8                                   20.8                                   20.8     Match
4                                      20.6                                   20.6                                   20.6     Match
5                                      20.3                                   20.4                                   20.4  No Match
6                                      20.4                                   20.5                                   20.5  No Match
7                                      21.4                                   21.4                                   21.4     Match
8                                      23.1                                   23.1                                   23.1     Match
9                                      24.9                                   25.0                                   25.0  No Match
10                                     26.6                                   26.7                                   26.7  No Match
11                                     27.9                                   27.9                                   27.9     Match
12                                     28.3                                   28.4                                   28.4  No Match
13                                     27.8                                   27.9                                   27.9  No Match
14                                     27.0                                   27.0                                   27.0     Match
15                                     27.3                                   27.4                                   27.4  No Match
16                                     26.4                                   26.5                                   26.5  No Match
17                                     25.1                                   25.1                                   25.1     Match
18                                     23.8                                   23.9                                   23.9  No Match
19                                     22.8                                   22.8                                   22.8     Match
20                                     22.0                                   22.0                                   22.0     Match
21                                     21.4                                   21.5                                   21.5  No Match
22                                     21.1                                   21.1                                   21.1     Match
23                                     20.8                                   20.9                                   20.9  No Match
24                                     20.9                                   20.9                                   20.9     Match
25                                     21.0                                   21.0                                   21.0     Match
26                                     21.0                                   21.0                                   21.0     Match
27                                     21.1                                   21.1                                   21.1     Match
28                                     21.2                                   21.3                                   21.3  No Match

     https://api.open-meteo.com_v1/dwd-icon  http://192.168.1.114:8080_v1/dwd-icon  http://192.168.1.110:8080_v1/dwd-icon     Match
0                                      20.9                                   21.0                                   21.0  No Match
1                                      20.8                                   20.9                                   20.9  No Match
2                                      20.9                                   20.9                                   20.9     Match
3                                      20.8                                   20.8                                   20.8     Match
4                                      20.6                                   20.6                                   20.6     Match
5                                      20.3                                   20.4                                   20.4  No Match
6                                      20.4                                   20.5                                   20.5  No Match
7                                      21.4                                   21.4                                   21.4     Match
8                                      23.1                                   23.1                                   23.1     Match
9                                      24.9                                   25.0                                   25.0  No Match
10                                     26.6                                   26.7                                   26.7  No Match
11                                     27.9                                   27.9                                   27.9     Match
12                                     28.3                                   28.4                                   28.4  No Match
13                                     27.8                                   27.9                                   27.9  No Match
14                                     27.0                                   27.0                                   27.0     Match
15                                     27.3                                   27.4                                   27.4  No Match
16                                     26.4                                   26.5                                   26.5  No Match
17                                     25.1                                   25.1                                   25.1     Match
18                                     23.8                                   23.9                                   23.9  No Match
19                                     22.8                                   22.8                                   22.8     Match
20                                     22.0                                   22.0                                   22.0     Match
21                                     21.4                                   21.5                                   21.5  No Match
22                                     21.1                                   21.1                                   21.1     Match
23                                     20.8                                   20.9                                   20.9  No Match
24                                     20.9                                   20.9                                   20.9     Match
25                                     21.0                                   21.0                                   21.0     Match
26                                     21.0                                   21.0                                   21.0     Match
27                                     21.1                                   21.1                                   21.1     Match

    https://api.open-meteo.com_v1/ecmwf  http://192.168.1.114:8080_v1/ecmwf  http://192.168.1.110:8080_v1/ecmwf     Match
0                                  19.2                                19.1                                19.1  No Match
1                                  18.2                                18.1                                18.1  No Match
2                                  18.8                                18.7                                18.7  No Match
3                                  25.9                                25.8                                25.8  No Match
4                                  30.0                                29.9                                29.9  No Match
5                                  29.1                                29.0                                29.0  No Match
6                                  23.7                                23.6                                23.6  No Match
7                                  20.7                                20.6                                20.6  No Match
8                                  20.1                                20.0                                20.0  No Match
9                                  19.0                                18.9                                18.9  No Match
10                                 19.1                                19.0                                19.0  No Match
11                                 26.4                                26.3                                26.3  No Match
12                                 30.2                                30.1                                30.1  No Match
13                                 26.9                                26.8                                26.8  No Match
14                                 23.1                                23.0                                23.0  No Match
15                                 20.8                                20.7                                20.7  No Match
16                                 19.0                                18.9                                18.9  No Match
17                                 17.4                                17.3                                17.3  No Match
18                                 17.5                                17.4                                17.4  No Match
19                                 24.8                                24.7                                24.7  No Match
20                                 29.2                                29.1                                29.1  No Match
21                                 26.7                                26.6                                26.6  No Match
22                                 22.4                                22.3                                22.3  No Match
23                                 20.5                                20.4                                20.4  No Match
24                                 18.9                                18.8                                18.8  No Match
25                                 18.7                                18.6                                18.6  No Match
26                                 18.5                                18.4                                18.4  No Match
27                                 24.5                                24.4                                24.4  No Match
28                                 29.4                                29.3                                29.3  No Match
29                                 28.2                                28.1                                28.1  No Match
30                                 22.5                                22.4                                22.4  No Match
31                                 21.2                                21.1                                21.1  No Match

Is there an explanation? Or how to ensure that the data correctly parsed fetched.

The logs contains a lot of errors but Patrick told me it's normal.

**Comment by patrick-zippenfenig** [comment]
Which version are running? There was a small change yesterday, that could effect rounding. I am working a binary format based on FlatBuffers which will return floating point values without rounding. Therefore there were larger changes to how data is rounded in text/ascii based formats

**Reply by meb427** [reply]
Currently using latest should be 0.2.69. ( deb package and docker )

**Reply by patrick-zippenfenig** [reply]
Upgrading to `0.2.86` should solve those differences. `0.2.69` is already 3 months old and there were plenty of bugfixes in the meantime :)

**Comment by meb427** [comment]
Also patrick noticed an issue in weather daily icons.
I was checking a location Tunis, tunisia   
 "latitude": 36.819,
 "longitude": 10.1658,
    
The daily icon here we get icon code 45 That is foggy according to WMO.
    
his is wrong.  As example you can see the day in yr.no
https://www.yr.no/en/forecast/hourly-table/2-2464470/Tunisia/Tunis%20Governorate/Tunis?i=2
    
That's true that most of the day is foggy, but over the night that don't count and in the end the day is synny. So day hours in this case would take over.

I'v found the algo that is supposed a bit like NOAA to do the logic 
    
DAILY:
If max probability is greater than 30% in any hour and total accumulation is greater than 1 mm, then precipitation type.
Type is based on the most common (modal) precipitation type.
If average visibility is less than 1 km, then fog.
If average wind speed is greater than 10 m/s, then wind.
If average cloud cover is greater than 75%, then cloudy.
If average cloud cover is greater than 37.5% and less than 75%, then partly-cloudy-day.
If average cloud cover is less than 37.5%, then clear.

But not sure what you are using?

May be need to focus on the day trend instead of the night? as the visibility don't count in the night.

**Reply by patrick-zippenfenig** [reply]
From my point of of the daily weather code should always reflect the worst weather condition. For your use case you should probably only take the maximum during 6:00-18:00. On my side I want to keep the data as "untainted" as possible.
